xmodule/docs/decisions/0004-restructure-video-block.rst

```diff
@@ -0,0 +1,101 @@
+
+Another crucial aspect is the rollout strategy. Given the platform's extensive user base, any changes to such a critical component must be introduced gradually to monitor performance and gather user feedback. The approach memo suggests the use of feature toggles to enable the new video player selectively. This will allow platform administrators to test the new features in a controlled environment before a full-scale rollout.
+
+Accessibility is a key focus area, and the new player aims to be compliant with the latest WCAG guidelines. This includes providing Audio Descriptions, focus state visibility, and other accessibility features, which are currently either lacking or not up to the mark in the existing system. The new architecture aims to rectify this by integrating these features into the core of the new Video Player.
+
+In summary, the context for this Architecture Decision Record (ADR) is introduction of the new Video XBlock aims to address multiple pain points, ranging from user experience and accessibility to extensibility and modernization of the tech stack. It represents a significant step forward in aligning the platform with modern web standards and user expectations, while also laying the groundwork for future enhancements.
+
+
+Decision
+--------
+
+1. **Move out current Video XBlock from xmodules and place it in a separate repository**
+
+   We have decided to extract the current Video XBlock from the xmodules directory and relocate it into its own dedicated repository within the github.com/openedx organization. This decision is driven by the need for improved modularity, maintainability, and the facilitation of continuous integration and deployment practices.
+   * A new repository will be created under the Open edX GitHub organization.
+   * The repository will adhere to Open edX's best practices, including code quality, testing, and documentation.
+   * The new XBlock will be hosted within the Open edX organization on GitHub, ensuring community involvement and adherence to Open edX standards.
+   * The proposed architecture for installing and using new xblock is the following:
+
+     .. image:: ./Architecture.png
+
+2. **React.js and Video.js Integration**
+   * React.js will be used to build a responsive, accessible frontend for a new Xblock as proposed in https://github.com/openedx/XBlock/issues/635 and prototyped in https://github.com/openedx/XBlock/issues/634.
+   * Video.js will be integrated for video playback, following the official guidelines (`Video.js Docs <https://docs.videojs.com/>`_).
+   * An NPM package will be created for the frontend, and it will be versioned and published (`Issue 635 <https://github.com/openedx/XBlock/issues/635>`_).
+
+3. **Feature Toggle in edx-platform**
+   * A feature toggle will be introduced using Django Waffle flags.
```

Comment:
If the implementation strategy is to move all the code out of the edx-platform repo, what would the feature toggle do? Is it purely the frontend rendering that's being toggled?

Reply:
If the migration strategy is purely frontend focused, e.g. the VideoBlock decides which way to render based on feature toggle, then I think that's fine, and extracting the code into a new repo before active development makes perfect sense. Then rollout could be done by course because the VideoBlock would always have the course context when deciding which version of its UI to render. Was that the idea?

If that's the case, and *if* the VideoBlock is going to keep around both sets of frontend assets, would it be lower effort to keep something closer to the old Studio experience for now, instead of the iframe editor there? Or at least, would it be practical to put off Studio changes until late in the process?

Reply:
You're correct, the feature toggle will be used to switch between legacy and new UI.
I will add the decision that user has an option for using either legacy editor or new editor in studio as it's already implemented now